app/controllers/users_controller.rb

```diff
@@ -12,14 +12,26 @@ def show
 
   def create
     @user = User.new(get_params)
-    if @user.save
-      AppMailer.send_welcome_message(@user).deliver
-      if params[:token]
-        invitation = Invitation.find_by_token(params[:token])
-        @user.follow(invitation.user)
-        invitation.user.follow(@user)
-        invitation.destroy
+    if @user.valid?
+
+      Stripe.api_key = ENV['stripe_api_key']
+      token = params[:stripeToken]
+      begin
+        charge = Stripe::Charge.create(
+          :amount => 999,
+          :currency => "usd",
+          :source => token,
+          :description => "sign-up fee for #{params[:user][:email]}"
+        )
+      rescue Stripe::CardError => e
+        flash[:danger] = "Your card has been declined: #{e}. Please try again."
+        render 'new' and return
       end
+
+      @user.save
+      AppMailer.send_welcome_message(@user).deliver
+      handle_invitation(params) if params[:invitation_token]

```

Comment:
Your methods in the controller have access to the `params` already, no need to pass them in.